modules/graceful/manager_unix.go

```diff
@@ -157,6 +158,7 @@ func (g *Manager) handleSignals(ctx context.Context) {
 		case <-ctx.Done():
 			log.Warn("PID: %d. Background context for manager closed - %v - Shutting down...", pid, ctx.Err())
 			g.DoGracefulShutdown()
+			break loop
```

Comment:
maybe just `return` is simpler?

Reply:
Does `break` has some shortcoming?

I think `break` is more clear because there are nested for+select. When the context is done, stopping the context loop seems more intuitive. And breaking out will make future code more easier to be added after the for loop, logs & further process.

Reply:
There is another `g.DoGracefulShutdown()` above but not break?

Reply:
No, that's different. 

The `g.DoGracefulShutdown()` above is triggered by `case sig := <-signalChannel:`, each signal triggers a call, there is no infinite loop.

But this one is trigger by `case <-ctx.Done():`, on a closed channel, it's a infinite loop.

Reply:
> Does `break` has some shortcoming?
> 
> I think `break` is more clear because there are nested for+select. When the context is done, stopping the context loop seems more intuitive. And breaking out will make future code more easier to be added after the for loop, logs & further process.

It's probably personal preference, result is the same. It's just a bit confusing for me to do a break there when it just exits anyway. I'm not sure if you need to optimize for future code that will or will not happen.

Reply:
in this case I would also prefer return, it would be more clear

Reply:
Yup, it's personal preference, result is the same. Since most people like `return` more, so there it is.